### PR TITLE
Add app-specific passwords feature

### LINKED ISSUE #7338: 2FA: Add app-specific password for API access
Currently the API is exempt of 2FA authentication because there isn't a standard way to handle TOTP token verification on mobile apps.

However, it would possible to emit app-specific passwords from the web interface protected by the 2FA token, and require one of them to obtain an OAuth token or with HTTP Auth from the API, disregarding even a valid account password when 2FA is enabled on the account.

This would effectively protect unwarranted app usage without apps having to support TOTP token verification.

Thanks @lostinlight for raising the issue!

### PR DESCRIPTION
**Pull request by MrPetovan**
Closes #7338

### CODE REVIEW

File: src/App/Router.php [hunk 1]
```diff
@@ -202,6 +202,7 @@ public function collectRoutes()
 			$collector->addGroup('/2fa', function (RouteCollector $collector) {
 				$collector->addRoute(['GET', 'POST'], '[/]'                    , Module\Settings\TwoFactor\Index::class);
 				$collector->addRoute(['GET', 'POST'], '/recovery'              , Module\Settings\TwoFactor\Recovery::class);
+				$collector->addRoute(['GET', 'POST'], '/app_specific'          , Module\Settings\TwoFactor\AppSpecific::class);
```
Comment (nupplaphil) [inline src/App/Router.php]: Wouldn't a route '2fa/app_specific' be more precise? Just asking :)

File: src/Module/Settings/TwoFactor/AppSpecific.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	{
+		if (!local_user()) {
+			return Login::form('settings/2fa/app_specific');
+		}
+
+		parent::content();
+
+		$appSpecificPasswords = AppSpecificPassword::getListForUser(local_user());
+
+		var_dump($appSpecificPasswords);
```
Comment (nupplaphil) [inline src/Module/Settings/TwoFactor/AppSpecific.php]: Seems like debug

File: src/App/Router.php (hunk 1 above)
Comment (MrPetovan) [reply src/App/Router.php]: The full route is `settings/2fa/app_specific`.

File: src/Module/Settings/TwoFactor/AppSpecific.php (hunk 2 above)
Comment (MrPetovan) [reply src/Module/Settings/TwoFactor/AppSpecific.php]: Oops.

File: src/App/Router.php (hunk 1 above)
Comment (nupplaphil) [reply src/App/Router.php]: Arg.. Didn't got that point on my mobile view.. Never mind..

**Pull request merged**